local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,340 @@
+"""Describes obstacles which the Sailbot must avoid: Boats and Land"""
+
+import math
+
+import numpy as np
+from custom_interfaces.msg import HelperAISShip
+from numpy.typing import NDArray
+from shapely.geometry import Point, Polygon
+
+from local_pathfinding.coord_systems import (
+    XY,
+    LatLon,
+    knots_to_kilometers_per_hour,
+    latlon_to_xy,
+    meters_to_km,
+)
+
+# Constants
+PROJ_TIME_NO_COLLISION = 3  # hours
+COLLISION_ZONE_SAFETY_BUFFER = 0.5  # km
+
+
+class Obstacle:
+
+    """
+    This Interface describes general obstacle objects which are
+    anything which the sailbot must avoid.
+
+    The choice was made to create an interface rather than a class as Boat and LandMass objects
+    may be very different objects which share an "avoidable" property and a position, but not much
+    else.
+
+    """
+
+    def __init__(
+        self,
+        reference: LatLon,
+        sailbot_position: LatLon,
+        sailbot_speed: float,
+    ):
+        """
+        Args:
+            -reference (LatLon): latitude and longitude of the next global waypoint
+            -sailbot_position (LatLon): lat and lon position of SailBot
+            -sailbot_speed (float): speed of the SailBot in km/h
+        """
+        self.reference = reference
+
+        # Position of the Sailbot in XY
+        self.sailbot_position = latlon_to_xy(reference, sailbot_position)
+
+        self.sailbot_speed = sailbot_speed
+
+        # This should be defined in child classes
+        self.collision_zone = Polygon(None, None)
+
+    def is_valid(self, reference: LatLon, point_latlon: LatLon) -> bool:
+        """
+        Checks if a point is contained the obstacle's interior
+
+        Args:
+            point (Point): a shapely Point object representing the point to be checked
+
+        Returns:
+            bool: True if the point is not within the obstacle's interior, False otherwise
+        """
+        point = latlon_to_xy(reference, point_latlon)
+
+        # contains() requires a shapely Point object as an argument
+        point = Point(point[0], point[1])
+        return not self.collision_zone.contains(point)
+
+
+class Boat(Obstacle):
+
+    """
+    This class describes boat objects which Sailbot must avoid.
+    Also referred to target ships or boat obstacles.
+
+
+    Attributes
+    ----------
+        -id (int): MMSI number of the boat
+        -speed_over_ground (float): speed of the boat, over ground, in knots
+        -collision_cone (Polygon): a shapely Polygon object representing the boat's collision box
```

Comment:
This should contain only attributes that are unique to `Boat`. Should move `collision_cone` to `Obstacle` attributes